exporters/variables-scss/src/content/generators/contentGenerator.ts

```diff
@@ -0,0 +1,57 @@
+import { Token, TokenGroup, TokenType } from '@supernovaio/sdk-exporters';
+import { exportConfiguration } from '../../index';
+import { generateCssFromTokens, generateCssObjectFromTokens } from './cssGenerator';
+import { formatCSS } from '../formatters/cssFormatter';
+
+// Add disclaimer to the top of the content
+const addDisclaimer = (content: string): string => {
+  if (exportConfiguration.generateDisclaimer) {
+    return `/* This file was generated by Supernova, don't change manually */\n${content}`;
+  }
+
+  return content;
+};
+
+export const generateFileContent = (
+  tokens: Token[],
+  mappedTokens: Map<string, Token>,
+  tokenGroups: Array<TokenGroup>,
+  tokenTypes: TokenType | TokenType[],
+  groupNames: string | string[],
+  withCssObject: boolean,
+  hasParentPrefix: boolean,
+) => {
+  let cssTokens = '';
+  let cssObject = '';
+
+  // If tokenTypes or groupNames are strings, convert them to arrays
+  const tokenTypesArray = Array.isArray(tokenTypes) ? tokenTypes : [tokenTypes];
+  const groupNamesArray = Array.isArray(groupNames) ? groupNames : [groupNames];
+
+  // Iterate over token types and group names to filter tokens
+  tokenTypesArray.forEach((tokenType) => {
+    groupNamesArray.forEach((group) => {
+      // Filter tokens by token type and group name
+      const filteredTokens = tokens.filter(
```

Comment:
```ts
const filterTokensByTypeAndGroup = (tokens: Token[], type: TokenType, group: string) =>
  tokens.filter((token) => token.tokenType === type && token.origin?.name?.includes(group));
```

Most of the time you write a comment it is a suggestion to encapsulate the logic into the named function :-)